### ISSUE TITLE
PHP8 + WP5.6 Compat

### ISSUE THREAD

**Issue by thomasplevy** [gocodebox/lifterlms-rest#209]
Add-on Checklist

+ Composer Updates
	+ lifterlms/lifterlms-tests
		+ [x] Switch from "dev-master" to "dev-trunk"
		+ [x] Update to latest version
	+ lifterlms/lifterlms-cs
		+ [x] Switch from "dev-master" to "dev-trunk"
		+ [x] Update to latest version
	+ Scripts
		+ [x] Update scripts to match the LifterLMS Core scripts block
		+ [x] Ensure any custom scripts are retained
		+ [x] Ensure database scripts use a custom default database name, eg "llms_groups_tests" instead of "llms_tests"
		+ [x] If the add-on is not configured for E2E testing the "env" and "env:setup" commands can be removed
	+ PHP Requirements
		+ [x] Update PHP version to be >=7.2

+ Travis Config Updates
	+ [x] Update to import the add-on.yml config from the LLMS core.
	+ [x] Import the e2e config if the add-on is setup for e2e testing
	+ [x] Retain any custom jobs specific to the add-on

+ Tests
	+ [x] Test Suite Passes in Travis
	+ [x] Passes manual inspection on php8

**Comment by eri-trabiccolo** [comment]
@thomasplevy 
I did this but I spend quite some time because there are to understand how to "fix" some issues.

Basically we're removing the lifterlms plugin install from the tests-install* scripts, and on the other hand we're installing the lifterlms plugin with Travis.

This means that you cannot really run tests locally because the lifterlms plugin is not installed.

On the other hand, if you re-add the plugin install in the tests-install* script, then the install will "break" because you know the lifterlms core plugin is already there and the cli asks what to do with the already existing files, and since nobody replies...

Shall we re-add the lifterlms plugin install in the composer.json and remove it from:
https://github.com/gocodebox/lifterlms/blob/437fe97231d6d74cfcd674aaa12d1aac23241a85/.config/travis/add-on.yml#L14
?

**Comment by thomasplevy** [comment]
> Shall we re-add the lifterlms plugin install in the composer.json and remove it from:
https://github.com/gocodebox/lifterlms/blob/437fe97231d6d74cfcd674aaa12d1aac23241a85/.config/travis/add-on.yml#L14
?

Yes, kind of. But it's more complicated because I very much want us to start testing add-ons against nightly builds of the core on the dev branch to give us advanced warning before shipping potential add-on incompatibilities.

You're right that we need to unify this so that the travis & local install commands both work (ideally identically), but if we do as you're proposing we lose the nightly build functionality that I built into travis.

I'd propose that we have the core install nightly installation *only* in travis, add the plugin installation back to the install script in composer and update the `plugin` command from `lifterlms-tests` to wipe the plugin directory prior to installing it.

In this way we can have easy local and travis installation and then when we specify a branch in travis (for the nightly builds from git) we can just overwrite the install of the core. There's a slight "performance" issue of installing the plugin twice during a nightly build but this is the easiest way I can think of to fix the issue I've created without losing a new feature that I think is important here.

What do you think?

**Comment by thomasplevy** [comment]
Fixes the prompt:

https://github.com/gocodebox/lifterlms-tests/commit/9440ec843aa1e3ab93cb156f7ee9bc6106649b1c

Here's a proposed composer config (relevant parts only):

```json
    "tests-install": [
      "\"vendor/bin/llms-tests\" install ${TESTS_DB_NAME:-llms_aq_tests} ${TESTS_DB_USER:-root} \"${TESTS_DB_PASS-password}\" ${TESTS_DB_HOST:-127.0.0.1} ${WP_VERSION:-latest} false \"${WP_TESTS_VERSION}\"",
      "@tests-install-plugins"
    ],
    "tests-install-php8": [
      "\"vendor/bin/llms-tests\" install ${TESTS_DB_NAME:-llms_aq_tests} ${TESTS_DB_USER:-root} \"${TESTS_DB_PASS-password}\" ${TESTS_DB_HOST:-127.0.0.1} ${WP_VERSION:-trunk} false trunk",
      "@tests-install-plugins"
    ],
    "tests-install-plugins": "\"vendor/bin/llms-tests\" plugin lifterlms",
    "tests-reinstall": [
      "@tests-remove",
      "@tests-install"
    ],
```

**Comment by eri-trabiccolo** [comment]
@thomasplevy 
~I think it's fine but, when on the travis install the re-install will happen for all the add-ons in any case, either we test against the nightly or not. Right?~

Ok saw the latest commit in core <3

**Comment by thomasplevy** [comment]
And this only installs from git when specified in travis:

https://github.com/gocodebox/lifterlms/commit/d5dbbfae38e2970baefce5fc3ce9e3da0b9977b9

**Comment by thomasplevy** [comment]
> I think it's fine but, when on the travis install the re-install will happen for all the add-ons in any case, either we test against the nightly or not. Right?

See my latest comment above @eri-trabiccolo

**Comment by eri-trabiccolo** [comment]
Yeah sorry @thomasplevy I edited my comment above when I saw the commit in core <3.

**Comment by eri-trabiccolo** [comment]
I think we can close this...

**Comment by thomasplevy** [comment]
@eri-trabiccolo DO IT THEN! I want you to have all the glory!

**Comment by eri-trabiccolo** [comment]
Closed with 8c64ec45b292c614f72d97d6549c2ac6a45adc52

**Comment by thomasplevy** [comment]
Oh yea
